### ISSUE TITLE
how to evaluate the Liberty using Notredame

### ISSUE THREAD

**Issue by hurricane2018** [DagnyT/hardnet#10]
![image](https://user-images.githubusercontent.com/36166137/38772957-7f2346a0-4010-11e8-9357-30d5c504ef6c.png)

The test accuracy is always change. Did you get the average precision of the test data ?

**Comment by hurricane2018** [comment]
I mean, If we train 10 epoch, we can get a precision. If we train 20 epoches, we can get another result. How did you solve this problem ?

**Comment by hurricane2018** [comment]
how many epoches did you use to evaluate a dataset ?

**Comment by DagnyT** [comment]
Hi! To get exactly same results from run to run, please, set up torch.backends.cudnn.deterministic = True, num_workers=0. 
We've trained for 10 epochs and evaluated results after the last one, also, take a look at the adjust_learning_rate function, which determines the learning rate policy.
If you have any other questions, don't hesitate to ask.

**Comment by DagnyT** [comment]
We didn't average the results during the evaluation. To get deterministic results from run to run, just add torch.backends.cudnn.deterministic = True to your code (check the last commit, I've added setup to get exactly same results from run to run).
